### ISSUE TITLE
[SCRAP]Apply ship range to discovery

### ISSUE THREAD

**Issue by CorrosiveSt** [CorrosiveSt/BROS_Demoproject#6]
Make the signal of either `sendSignal_HL()` or `sendSignal_DC()` bounce to the next tile as the range needs

**Comment by CorrosiveSt** [comment]
I could build a simplified table format from `curr_HL()` Which would list the positions of the last tiles:
```lua
range_bounce_tiles = {
  {Vector position}, -- [1]
  {Vector position}, -- [2]
  {Vector position}, -- [3]
  {Vector position}, -- [4]
  ...
}
```
Then use a `for` loop to iterate through the positions, and calling sendPulse(range). This way the range will stay consistant even thought the function is called for individual objects.

**Comment by CorrosiveSt** [comment]
I decided to use the already present range_bounce table, which just inserts the last pieces of any cast table. I might want to simplify that code somehow, maybe even having another argument inside for object reference (Though it seems like it's doing that quite well already)

**Comment by CorrosiveSt** [comment]
This shit is impossible to debug... I'm just gonna simplify the function now, and come back to this

**Comment by CorrosiveSt** [comment]
After losing my humanity, I decided to detach BOTH the validation process AND the physics casting to their own functions. This should ease readibility, and make changes easier to do in the future.

**Comment by CorrosiveSt** [comment]
brrt I am an airplane agagaga goo goo

**Comment by ronip-mc** [comment]
Update log #31521
Subject "Corrosive" seems to have been released from his own chair and now has lost control over himself, further medical tests required and security staff have been dispatched on scene, further reports will be written once the subject comes under control once more.

Update log #32620
Subject "Corrosive" has seen himself in the next 10 years as an "Immediate Java Employee" and nothing else, this subject has been lost to all matters and is unrecoverable, research on this subject has been called off as subject is now of no help towards the research, thus he has been released into the wild, to be where he was destined.
Report End.

**Comment by CorrosiveSt** [comment]
No matter what I try, the casts are either:
- Hurting the performance without sanity/range checking
- Moving in unpredictable and non-logical patterns

Hypothesis: It really doesn't like writing in tables that have already been added by itself, although this also could be just me fucking up the order of discovery/range checks.

I'm gonna try to move the range checking to its own function (`rangeCheck()`), and possibly go from there.

EDIT: It did not work at all, so the problem doesn't exist within `sendPulse()`. Next up would be to somehow compile my own table, with every range-tagged pulse having its own table inside the previous one. Either way I somehow need the pulses to sync up, no matter the performance cost.

**Comment by CorrosiveSt** [comment]
New plan, use sub-tables to index through all detected tiles. After one pulse, `sendPulse()` will call `rangeCheck()`, which will calculate the needed range. If the range is still valid, index the latest added table, and iterate through the whole thing, passing `sendPulse()` to the given tiles _(Maybe with a notable delay in between for future's sake?)_. 
![Blank diagram](https://user-images.githubusercontent.com/65394719/165517679-c8c10730-e139-4ceb-8757-1dfdaff76c1c.png)

This will create a subtable system, where the main cast tile would only serve as a container for the rest. This way we could reset the subtable betwen every new pulse, which originate > 2 tiles away _(See diagram)_.

EDIT: I realised I could follow this logic easier consistantly if I just issue the start_pos as a **table of positions** instead of looping through each one. This way I can intergrate that into the fetching script AND have a slight delay between the pulses anyways, so it'll *maybe* be performant enough. The process therefore would work like:
1. Call sendPulse_HL with the singular start pos parameter
2. The now present for loop in the verification process will just use that given value, all good here
3. When the time comes to check the available range, check range, and pass the latest batch of coordinates back to sendPulse_HL
4. This way I can more easily organize the starting positions for each pulse

**Comment by CorrosiveSt** [comment]
Adding to previous, I've also decided to reformat the output (OR returned) table, which will house the tiles for an *Individual* cast. This way I can return the singular table, which will contain the tables made in itself, then refer to those tables using the dot parameter.
Example usage:
```lua
cast_tiles = {
  space = {tile1, tile2, tile3},
  sw = {tile1, tile2, tile3}
}
```
EDIT: And while this does mean that object parameters for individual objects will have to be referred to something like `cast_tiles.space[5].getPosition()`, I honestly think the 1-table system will benefit the project overall

**Comment by CorrosiveSt** [comment]
So close to a win here, yet so far away...
If I try to pulse everything at once, the game freezes and kills my ram.
If I try to space it out, the timings won't match, and it'll pulse everything and nothing that it needs to.
I can either try:
1. Get the code to wait for a return code before making another pulse. I need further testing to see if that's somehow possible without adding another Wait.frames() timer, or if I could make it return only AFTER it has hit the second Space tile.
2. Making a queue system, where it'll leave out time for the next pulse depending on the ship's discovery range (since the amount of needed pulses increases VERY fast with range.)
3. Give up, cry, and use a physical object to detect the pulses. I don't know if that'll make things worse, as it takes a longer time for the objects to get to the bouncepoint.

**Comment by CorrosiveSt** [comment]
![Untitled Diagram drawio(3)](https://user-images.githubusercontent.com/65394719/184530099-190c561a-abdc-487e-bb71-fae2d549072c.png)


Decided to go with the queue system, hopefully possible to implement...

Edited to include background color

**Comment by ronip-mc** [comment]
Okay but hear me out, what if instead of multiple pulses we will rework the tiles as such that there will only be 1 pulse which takes a number of turns to do, which should be easier for you to accomplish, whilst also removing all lane tiles and switching it to be continuous tiles, even empty ones, which you can travel to
- from fartship explorer

**Comment by CorrosiveSt** [comment]
Elaborate in the projects tab. Closing for reason: "Fuck this".
